spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainerIntegration2Tests.java

```diff
@@ -475,6 +468,34 @@ public void onMessage(Message message) {
 		container.get().stop();
 	}
 
+	@Test
+	public void testTransientBadMessageDoesntStopContainer() throws Exception {
+		CountDownLatch latch = new CountDownLatch(3);
+		this.container = createContainer(new MessageListenerAdapter(new PojoListener(latch, false)), this.queue.getName());
+		this.template.convertAndSend(this.queue.getName(), "foo");
+		this.template.convertAndSend(this.queue.getName(), new Foo());
+		this.template.convertAndSend(this.queue.getName(), new Bar());
+		this.template.convertAndSend(this.queue.getName(), "foo");
+		assertTrue(latch.await(10, TimeUnit.SECONDS));
+		assertTrue(this.container.isRunning());
+		this.container.stop();
+	}
+
+	@Test
+	public void testTransientBadMessageDoesntStopContainerLambda() throws Exception {
+		final CountDownLatch latch = new CountDownLatch(2);
+		this.container = createContainer(new MessageListenerAdapter((ReplyingMessageListener<String, Void>) m -> {
```

Comment:
I guess this code can't be cherry-picked.
NP. Will fix during backport 
:smile: